src/graphics/program-lib/chunks/standard/frag/normalDetailMap.js

```diff
@@ -7,17 +7,17 @@ vec3 blendNormals(vec3 n1, vec3 n2) {
     // https://blog.selfshadow.com/publications/blending-in-detail/#detail-oriented
     n1 += vec3(0, 0, 1);
     n2 *= vec3(-1, -1, 1);
-    return normalize(n1*dot(n1, n2)/n1.z - n2);
+    return n1 * dot(n1, n2) - n2 * n1.z;
 }
 #endif
 
 vec3 addNormalDetail(vec3 normalMap) {
-    #ifdef MAPTEXTURE
+#ifdef MAPTEXTURE
     vec3 normalDetailMap = unpackNormal(texture2D(texture_normalDetailMap, $UV, textureBias));
-    normalDetailMap = normalize(mix(vec3(0.0, 0.0, 1.0), normalDetailMap, material_normalDetailMapBumpiness));
+    normalDetailMap = mix(vec3(0.0, 0.0, 1.0), normalDetailMap, material_normalDetailMapBumpiness);
```

Comment:
is either the source detail normal, or the output of the blendNormals, normalized somewhere?

Reply:
Yes the final blended normal is now being normalised post TBN transform (it was being normalised pre-TBN before). See normalMap.js line 11.